File: src/main/java/com/alan/enums/MyEnum.java
package com.alan.enums;

/**
 * @author mengqinghao
 * @Title: myFunctions
 * @Package com.alan.enums
 * @Description: ${TODO}
 * @date 2018/4/8下午3:27
 */
public enum MyEnum {
    A("a"),B("b"),C("c");
    private String value;

    private MyEnum(String value){
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value ;
    }
}

File: src/main/java/com/alan/enums/MyEnum.kt
package com.alan.enums

/**
 * @author mengqinghao
 * @Title: myFunctions
 * @Package com.alan.enums
 * @Description: ${TODO}
 * @date 2018/4/8下午3:27
 */
enum class MyEnum private constructor(private val value: String) {
    A("a"), B("b"), C("c");

    override fun toString(): String {
        return this.value
    }
}
